### PR TITLE
feat(tenant-management): integrate multiple idps

### PR DESCRIPTION
**Pull request by Tyagi-Sunny**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/tenant-management-service/src/models/dtos/idp-details-dto.model.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@model({
+  description: 'model describing payload for IDP controller',
+})
+export class IdpDetailsDTO extends Model {
+  @property({
+    type: 'object',
+    description: 'Tenat object',
+    jsonSchema: getJsonSchema(Object),
+  })
+  tenant: AnyObject;
```
Comment (Tyagi-Sunny) [inline services/tenant-management-service/src/models/dtos/idp-details-dto.model.ts:15]: why are we giving tenant as AnyObject

File: services/tenant-management-service/src/models/dtos/tenant-dto.model.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import {model, property} from '@loopback/repository';
+import {Address} from '../address.model';
+
+import {Tenant} from '../tenant.model';
+import {Contact} from '../contact.model';
+
+@model({
+  description: 'model describing payload used to create a lead',
+})
+export class TenantDto extends Tenant {
```
Comment (Tyagi-Sunny) [inline services/tenant-management-service/src/models/dtos/tenant-dto.model.ts:11]: made changes to TenantDto and set this as type of tenant in IdpDetailsDto

File: services/tenant-management-service/src/providers/idp/types.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export type Auth0Response = {
```
Comment (Tyagi-Sunny) [inline services/tenant-management-service/src/providers/idp/types.ts:1]: as we are using a commong resp IdpResp={authId:string}, we can remove this Auth)response

File: services/tenant-management-service/src/webhook.component.ts [hunk 4]
```diff
@@ -126,12 +137,24 @@ export class WebhookTenantManagementServiceComponent implements Component {
...
       WebhookDTO,
+      TenantConfig,
     ];
 
-    this.controllers = [WebhookController];
+    this.controllers = [
+      WebhookController,
+      IdpController,
+      TenantConfigController,
+      TenantConfigTenantController,
```
Comment (Tyagi-Sunny) [inline services/tenant-management-service/src/webhook.component.ts:147]: remove tenantConfigController and tenantconfigtenat controller from here. only idp controller should be her

File: services/tenant-management-service/src/component.ts [hunk 5]
```diff
@@ -146,6 +154,9 @@ export class TenantManagementServiceComponent implements Component {
       LeadController,
       PingController,
       TenantController,
+      IdpController,
```
[the remark this answers is not in the document]
Comment (Tyagi-Sunny) [reply services/tenant-management-service/src/component.ts]: done

File: services/tenant-management-service/src/providers/idp/idp-auth0.provider.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+          // eslint-disable-next-line
+          page_background: configValue.page_background,
+        },
+      },
+      // eslint-disable-next-line
+      enabled_connections: configValue.enabled_connections,
+    };
+    function generateStrongPassword(length: number): string {
+      const charset =
+        'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()_+~`|}{[]:;?><,./-=';
```
Comment (yeshamavani) [inline services/tenant-management-service/src/providers/idp/idp-auth0.provider.ts:61]: use regex @Surbhi-sharma1

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: services/tenant-management-service/src/providers/idp/idp-auth0.provider.ts (hunk 6 above)
Comment (Surbhi-sharma1) [reply services/tenant-management-service/src/providers/idp/idp-auth0.provider.ts]: Done

**Pull request merged**